pkg/upgrade/upgrades/upgrades.go

```diff
@@ -132,6 +132,14 @@ var upgrades = []upgradebase.Upgrade{
 		upgrade.RestoreActionNotRequired("cluster restore does not restore this table"),
 	),
 
+	upgrade.NewTenantUpgrade(
+		"delete all-tenants version row in system.tenant_settings",
+		clusterversion.V24_2_DeleteAllTenantSettingsVersion.Version(),
+		upgrade.NoPrecondition,
+		deleteAllTenantsVersionTenantSettings,
+		upgrade.RestoreActionNotRequired("TODO"),
```

Comment:
We could remove this erroneous value on restore in `tenantSettingsTableRestoreFunc`.

Reply:
Pushed something. Where are changes like this tested?

Reply:
The easiest way is probably to add a new  backup artifact in

```
pkg/ccl/backupccl/testdata/restore_old_versions/
```

You'd create a cluster with the "bad" state and then create a backup for it to `nodelocal://` then you can copy the backup files into a directory there and then modify 
`pkg/ccl/backupccl/restore_old_versions_test.go` to include your new test case.

Reply:
Thanks, done.